song_rater/views.py

```diff
@@ -13,5 +15,30 @@ class SongCreate(CreateView):
     success_url = '/song-added'
 
 
+class SongList(ListView):
+    model = Song
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        song_table = []
+
+        for song in Song.objects.all().prefetch_related('ratings'):
+            ratings = song.ratings.all()
+            n_ratings = ratings.count()
+            avg_rating = ratings.aggregate(Avg('rating'))['rating__avg'] or '-'
+
+            song_table.append({
+                'artist': song.artist,
+                'title': song.title,
+                'avg_rating': avg_rating,
+                'n_ratings': n_ratings,
```

Comment:
I think you can skip the introduction of the `n_ratings` variable.  `ratings.count()` is just as clear.  
Just stick the expresion you evaluate directly into the dict. :) 

The same might be true for 'avg_rating'

Reply:
Good find! Thank you